tests/commands/test_cache.py

```diff
@@ -59,30 +57,36 @@ def test_rm_add_update(tmp_library: TemporaryLibrary) -> None:
     query_results = db.query_dict({"title": title})
     assert len(query_results) == 1
 
-    # udpate
+    # update
     ###################################################
-    doc["cache"] = "testing"
-    doc.save()
+    # NOTE: modifying `doc` diretly may modify the version in the database, so
+    # this modifies the info file behind its back completely to check the update
+    doc_dict = {**dict(doc), "cache": "test-update"}
+    papis.yaml.data_to_yaml(doc.get_info_file(), doc_dict, allow_unicode=True)
 
-    query_results = db.query_dict({"cache": "testing"})
+    query_results = db.query_dict({"cache": "test-update"})
     assert len(query_results) == 0
 
     result = cli_runner.invoke(cli, ["update", "--doc-folder", folder])
     assert result.exit_code == 0
 
-    query_results = db.query_dict({"cache": "testing"})
+    query_results = db.query_dict({"cache": "test-update"})
     assert len(query_results) == 1
 
     # update-newer
     ###################################################
-    doc["cache"] = "end-testing"
-    doc.save()
+    # NOTE: `update-newer` checks mtimes, so we wait a bit so that the mtime is
+    # meaningfully changed and the OS had time to update it
+    time.sleep(1)
+
+    doc_dict = {**dict(doc), "cache": "test-update-newer"}
+    papis.yaml.data_to_yaml(doc.get_info_file(), doc_dict, allow_unicode=True)
 
-    query_results = db.query_dict({"cache": "end-testing"})
+    query_results = db.query_dict({"cache": "test-update-newer"})
     assert len(query_results) == 0
 
     result = cli_runner.invoke(cli, ["update-newer", "--all"])
     assert result.exit_code == 0
 
-    query_results = db.query_dict({"cache": "end-testing"})
+    query_results = db.query_dict({"cache": "test-update-newer"})
     assert len(query_results) == 1
```

Comment:
@alejandrogallo I managed to hit this error on my Linux machine too. As far as I can tell, it seems to have just been an issue with `st_mtime` not catching the changes fast enough.

Adding a little `time.sleep(1)` gives an updated `mtime` and makes the test pass. There's also a `st_mtime_ns` in nanoseconds that we could use here, if it becomes an issue.